context.go

```diff
@@ -96,3 +123,77 @@ func (c *Context) Service(name string) Service {
 func (c *Context) String() string {
 	return c.conode.ServerIdentity.String()
 }
+
+var contextData = map[string][]byte{}
+
+// Save takes an identifier and an interface. The interface will be network.Marshaled
+// and written to the filename based on the identifier. An eventual error will be returned.
+func (c *Context) Save(id string, data interface{}) error {
+	buf, err := network.MarshalRegisteredType(data)
+	if err != nil {
+		return err
+	}
+	fname := c.Path(id)
+	if contextDataPath == "" {
+		contextData[fname] = buf
+		return nil
+	}
+	return ioutil.WriteFile(fname, buf, 0660)
```

Comment:
Permissions should be specified within the comments, and I'd put it as 0640 or stg. I don't know it's a delicate subject...